File: epi_judge_java/epi/strings/IsStringPalindromic.java
package epi.strings;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class IsStringPalindromic {
    @EpiTest(testDataFile = "is_string_palindromic.tsv")

    public static boolean isPalindromic(String s) {
        int i1 = 0, i2 = s.length() - 1;
        while (i1 < i2) {

            while (!Character.isLetter(s.charAt(i1)) && i1<i2) i1++;
            while (!Character.isLetter(s.charAt(i2)) && i1<i2) i2--;

            if (s.charAt(i1) != s.charAt(i2)) return false;
        }

        return true;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "IsStringPalindromic.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_java/epi/strings/kt/IsStringPalindromic.kt
package epi.strings.kt

import epi.test_framework.EpiTest
import epi.test_framework.GenericTest

object IsStringPalindromic {
    @JvmStatic
    @EpiTest(testDataFile = "is_string_palindromic.tsv")
    fun isPalindromic(s: String?): Boolean {
        if (s == null) return false;
        var i1 = 0;
        var i2 = s.length - 1;

        while (i1 < i2) {

            while (!Character.isLetter(s[i1]) && i1 < i2) i1++;
            while (!Character.isLetter(s[i2]) && i1 < i2) i2--;

            if (Character.toLowerCase(s[i1].toLowerCase()) != Character.toLowerCase(s[i2])) return false;
        }
        return true
    }

    @JvmStatic
    fun main(args: Array<String>) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "IsStringPalindromic.java",
                                object : Any() {}.javaClass.enclosingClass)
                        .ordinal)
    }
}